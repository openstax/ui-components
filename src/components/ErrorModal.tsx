import { Button } from "./Button";
import { Modal, ModalFooter, ModalPropTypes } from "./Modal";
import { Error } from "./Error";

type ErrorModalProps = React.PropsWithChildren<Omit<ModalPropTypes, 'variant' | 'heading'>>;

export const ErrorModal = (props: ErrorModalProps) => {
  return <Modal {...props} variant='error' heading='Error'>
    <Error />
    <ModalFooter><Button onClick={props.onModalClose}>OK</Button></ModalFooter>
  </Modal>;
};
