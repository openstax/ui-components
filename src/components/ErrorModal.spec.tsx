import renderer from 'react-test-renderer';
import { ErrorModal } from './ErrorModal';

describe('Modal', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <ErrorModal onModalClose={jest.fn()} show={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
