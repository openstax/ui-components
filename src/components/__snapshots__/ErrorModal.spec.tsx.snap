// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
<body>
  <main
    aria-hidden="true"
    id="root"
  />
  <span
    aria-hidden="true"
    data-focus-scope-start="true"
    hidden=""
  />
  <div
    class="sc-papXJ dSNZGe"
    data-rac=""
    style="--visual-viewport-height: 768px;"
  >
    <div
      class="react-aria-Modal"
      data-rac=""
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <button
          aria-label="Dismiss"
          id="react-aria-1"
          style="width: 1px; height: 1px;"
          tabindex="-1"
        />
      </div>
      <div
        class="sc-jqUVSM kHnGYC"
      >
        <section
          aria-labelledby="react-aria-2"
          class="sc-jSMfEi jOuQOj"
          data-rac=""
          role="dialog"
          tabindex="-1"
        >
          <header
            class="sc-gKXOVf fMZXQS"
          >
            <h2
              class="sc-iBkjds crmGxr"
              id="react-aria-2"
              slot="title"
            >
              Error
            </h2>
            <button
              aria-label="Close"
              class="sc-eCYdqJ dRnxZp"
              type="button"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                height="15px"
                version="1.1"
                viewBox="0 0 15 15"
                width="15px"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill="currentColor"
                    transform="translate(-302.000000, -18.000000)"
                  >
                    <g
                      transform="translate(302.000000, 18.000000)"
                    >
                      <path
                        d="M7.5,5.41522791 L12.0331524,0.579865364 C12.3077536,0.286957429 12.7165503,0.24816296 12.946282,0.493210121 L13.9861449,1.60239723 C14.2158766,1.84744439 14.1795068,2.28349422 13.9049056,2.57640216 L9.37175324,7.41176471 L13.9049056,12.2471273 C14.1795068,12.5400352 14.2158766,12.976085 13.9861449,13.2211322 L12.946282,14.3303193 C12.7165503,14.5753665 12.3077536,14.536572 12.0331524,14.243664 L7.5,9.4083015 L2.96684761,14.243664 C2.69224642,14.536572 2.2834497,14.5753665 2.05371799,14.3303193 L1.01385508,13.2211322 C0.784123363,12.976085 0.820493178,12.5400352 1.09509437,12.2471273 L5.62824676,7.41176471 L1.09509437,2.57640216 C0.820493178,2.28349422 0.784123363,1.84744439 1.01385508,1.60239723 L2.05371799,0.493210121 C2.2834497,0.24816296 2.69224642,0.286957429 2.96684761,0.579865364 L7.5,5.41522791 Z"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </button>
          </header>
          <div
            class="sc-evZas eSCdIH"
            data-testid="error"
          >
            <h3
              class="sc-crXcEl RjTbW"
            >
              Uh-oh, there's been a glitch
            </h3>
            We're not quite sure what went wrong. Restart your browser. If this doesn't solve the problem, visit our 
            <a
              href="https://openstax.secure.force.com/help"
              target="_blank"
            >
              Support Center
            </a>
            .
            <div
              class="sc-breuTD feIzvt"
              data-testid="event-id"
            />
          </div>
          <div
            class="sc-kDDrLX elkCOb"
          >
            <button
              class="sc-bczRLJ dOOknK"
            >
              OK
            </button>
          </div>
        </section>
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-focus-scope-end="true"
    hidden=""
  />
</body>
`;

exports[`Modal matches snapshot when heading and children are set 1`] = `
<body>
  <main
    aria-hidden="true"
    id="root"
  />
  <span
    aria-hidden="true"
    data-focus-scope-start="true"
    hidden=""
  />
  <div
    class="sc-papXJ dSNZGe"
    data-rac=""
    style="--visual-viewport-height: 768px;"
  >
    <div
      class="react-aria-Modal"
      data-rac=""
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <button
          aria-label="Dismiss"
          id="react-aria-3"
          style="width: 1px; height: 1px;"
          tabindex="-1"
        />
      </div>
      <div
        class="sc-jqUVSM kHnGYC"
      >
        <section
          aria-labelledby="react-aria-4"
          class="sc-jSMfEi jOuQOj"
          data-rac=""
          role="dialog"
          tabindex="-1"
        >
          <header
            class="sc-gKXOVf fMZXQS"
          >
            <h2
              class="sc-iBkjds crmGxr"
              id="react-aria-4"
              slot="title"
            >
              Error
            </h2>
            <button
              aria-label="Close"
              class="sc-eCYdqJ dRnxZp"
              type="button"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                height="15px"
                version="1.1"
                viewBox="0 0 15 15"
                width="15px"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill="currentColor"
                    transform="translate(-302.000000, -18.000000)"
                  >
                    <g
                      transform="translate(302.000000, 18.000000)"
                    >
                      <path
                        d="M7.5,5.41522791 L12.0331524,0.579865364 C12.3077536,0.286957429 12.7165503,0.24816296 12.946282,0.493210121 L13.9861449,1.60239723 C14.2158766,1.84744439 14.1795068,2.28349422 13.9049056,2.57640216 L9.37175324,7.41176471 L13.9049056,12.2471273 C14.1795068,12.5400352 14.2158766,12.976085 13.9861449,13.2211322 L12.946282,14.3303193 C12.7165503,14.5753665 12.3077536,14.536572 12.0331524,14.243664 L7.5,9.4083015 L2.96684761,14.243664 C2.69224642,14.536572 2.2834497,14.5753665 2.05371799,14.3303193 L1.01385508,13.2211322 C0.784123363,12.976085 0.820493178,12.5400352 1.09509437,12.2471273 L5.62824676,7.41176471 L1.09509437,2.57640216 C0.820493178,2.28349422 0.784123363,1.84744439 1.01385508,1.60239723 L2.05371799,0.493210121 C2.2834497,0.24816296 2.69224642,0.286957429 2.96684761,0.579865364 L7.5,5.41522791 Z"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </button>
          </header>
          <div
            class="sc-evZas eSCdIH"
            data-testid="error"
          >
            <h3
              class="sc-crXcEl RjTbW"
            >
              Scores not sent
            </h3>
            There was an issue syncing your scores. Please try again in a few minutes. If the issue persists, visit our
            <a
              href="https://openstax.secure.force.com/help"
              rel="noreferrer"
              target="_blank"
            >
              Support Center
            </a>
            .
            <div
              class="sc-breuTD feIzvt"
              data-testid="event-id"
            />
          </div>
          <div
            class="sc-kDDrLX elkCOb"
          >
            <button
              class="sc-bczRLJ dOOknK"
            >
              OK
            </button>
          </div>
        </section>
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-focus-scope-end="true"
    hidden=""
  />
</body>
`;
